skimage/exposure/exposure.py

```diff
@@ -346,6 +346,14 @@ def rescale_intensity(image, in_range='image', out_range='dtype'):
     imin, imax = intensity_range(image, in_range)
```

Comment:
The check for nans may be implemented only for `imin` an `imax` using
```python
np.isnan([imin, imax]).any()
```
If true, the user should be warned that the input image contains nans and that this causes the return array to be full of nans.  You may then directly return `np.full_like(image, np.nan)`

Reply:
:-) I am sorry if my comment was not clear enough..
I mean that you don't need to check if `omax` and `omin` are nans and you can early stop the algorithm by returning `np.full_like(image, np.nan)` to simply prevents from doing unnecessary computations...

Reply:
No problem. Actually we do need to check `omin` and `omax` as well. E.g. if `out_range="image"` they can be NaN as well. 

I'd like to avoid extra code paths if possible. And I consider cases were NaN is broadcasted to the full image as undesired behavior most of the time. Unnecessary computations don't matter then.

Reply:
I only considered the case where `in_range` is `image`, here is my mistake, sorry.